docs/enhancements/add-feature-support-by-generic-attributes.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Add feature-support-level by generic attributes
+authors:
+- "@eliorerz"
+  creation-date: 2023-02-23
+  last-updated: 2023-03-01
+---
+
+# Add feature-support-level by a generic attributes
+
+## Summary
+
+In the current implementation of Assisted Installer, there is an API that allows the user to get a list of objects
+containing a list of features attached to an OpenShift version and its support-level.
+Whenever a user wants to know if a feature is supported in an OpenShift version, the user needs to call
+the feature-support-list API, get the entire list and filter the results to find the specific feature for that version.
+As part supporting multiple CPU architectures on Assisted Installer, we want the ability to add another dimension to the
+returned object - CPU architecture, so we can determine if a feature is supported or not by two parameters instead of one.
+
+
+## Motivation
+Adding another attribute to the feature-support-level API is indeed required but in its current implementation there are
+few flaws (from my perspective):
+1. The current implementation forces us to duplicate the entire features list for each cpu architecture (API change).\
+   For example if for `OpenshiftVersion = 4.12` we currently hold 10 features so adding just ARM64 as a CPU architecture to
+   that equation will require us to duplicate the 10 features (for that architecture) and set the support-level for both
+   x86 and arm64.
```

Comment:
maybe even mentioning there are even more archs than that (``s390x`` & ``ppc64le``) so it's really is too much to just duplicate entries